hikari/api/chunker.py

```diff
@@ -38,14 +161,168 @@ class GuildChunker(abc.ABC):
     __slots__: typing.Sequence[str] = ()
 
     @abc.abstractmethod
-    async def request_guild_chunk(self, guild: guilds.GatewayGuild) -> None:
+    def fetch_members_for_guild(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.GatewayGuild],
+        *,
+        timeout: typing.Union[int, float, None],
+        limit: typing.Optional[int],
+        include_presences: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        query_limit: int = 0,
+        query: str = "",
+        users: undefined.UndefinedOr[typing.Sequence[snowflakes.SnowflakeishOr[users_.User]]] = undefined.UNDEFINED,
+    ) -> event_stream.Streamer[shard_events.MemberChunkEvent]:
+        """Request for a guild chunk.
+
+        Parameters
+        ----------
+        guild: hikari.guilds.Guild
+            The guild to request chunk for.
+        timeout: builtins.int or builtins.float or builtins.None
+            The maximum amount of time the returned stream should spend waiting
+            for the next chunk event to be received before ending the iteration.
+            If `builtins.None` then this will never timeout between events.
+        limit: builtins.int or builtins.None
+            The limit for how many events the streamer should queue before
+            dropping extra received events. Leave as `builtins.None` for this to
+            be unlimited.
+        include_presences: hikari.undefined.UndefinedOr[builtins.bool]
+            If specified, whether to request presences.
+        query: builtins.str
+            If not `builtins.None`, request the members which username starts with the string.
+        query_limit: builtins.int
+            Maximum number of members to send matching the query.
+        users: hikari.undefined.UndefinedOr[typing.Sequence[hikari.snowflakes.SnowflakeishOr[hikari.users.User]]]
+            If specified, the users to request for.
+
+        !!! note
+            To request the full list of members, set `query` to `builtins.None` or `""`
+            (empty string) and `limit` to 0.
+
+        !!! note
+            The chunk request will not be sent off until the returned stream is
+            opened.
+
+        !!! warning
+            Validation errors like `builtins.ValueError` and
+            `hikari.errors.MissingIntentError` will be delayed until you open
+            the returned stream.
+
+        Returns
+        -------
+        hikari.utiltiies.event_stream.Streamer[hikari.events.shard_events.MemberChunkEvent]
+            A stream of chunk events for the generated request.
+        """
+
+    @abc.abstractmethod
+    async def get_request_status(self, nonce: str) -> typing.Optional[RequestInformation]:
+        """Return the status of a request.
+
+        Parameters
+        ----------
+        nonce: str
```

Comment:
these colons should start with spaces right

Reply:
Not sure, the lib's inconsistently using both styles rn

Reply:
it shouldn't be, cause `:` without a preceding space broke pdoc3 last i checked

Reply:
i don't ever remember one style breaking pdoc3